openapiart/bundler.py

```diff
@@ -939,18 +1013,27 @@ def _apply_common_x_field_pattern_properties(
         if "default" in xpattern:
             schema["default"] = xpattern["default"]
             if property_name == "step":
-                if format in step_defaults:
-                    schema["default"] = step_defaults[format]
+                if fmt in step_defaults:
+                    schema["default"] = step_defaults[fmt]
                 else:
                     schema["default"] = 1
             elif property_name == "values":
                 schema["default"] = [schema["default"]]
-        if format is not None:
+
+        finalised_format = None
+        if xpattern["format"] == "integer":
+            finalised_format = "uint32"
+            if "length" in xpattern and xpattern["length"] > 32:
+                finalised_format = "uint64"
+        elif fmt is not None:
+            finalised_format = fmt
+
+        if finalised_format is not None:
             # TODO: fix this
-            # if property_name == "values":
-            #     schema["items"]["format"] = format
-            # else:
-            schema["format"] = format
+            if property_name == "values":
+                schema["items"]["format"] = finalised_format
+            else:
+                schema["format"] = finalised_format
         if "length" in xpattern:
```

Comment:
when are we setting min/max for items in values ?

Reply:
done